usecase/parser/ibcmsg/ibcmsg.go

```diff
@@ -1251,3 +1251,559 @@ func ParseMsgTimeoutOnClose(
 		params,
 	)}
 }
+
+func ParseMsgGrant(
+	msgCommonParams event.MsgCommonParams,
+	msg map[string]interface{},
+) []command.Command {
+	authType := msg["grant"].(map[string]interface{})["authorization"].(map[string]interface{})["@type"]
+
+	switch authType {
+	case "/cosmos.bank.v1beta1.SendAuthorization":
+		return parseRawMsgSendGrant(msgCommonParams, msg)
+	case "/cosmos.bank.v1beta1.StakeAuthorization":
+		return parseRawMsgStackGrant(msgCommonParams, msg)
+	case "/cosmos.bank.v1beta1.GenericAuthorization":
+		return parseRawMsgGenericGrant(msgCommonParams, msg)
+	default:
+		return []command.Command{}
+	}
+}
+
+func parseRawMsgSendGrant(
+	msgCommonParams event.MsgCommonParams,
+	msg map[string]interface{},
+) []command.Command {
+	var rawMsg ibc_model.RawMsgSendGrant
+	decoderConfig := &mapstructure.DecoderConfig{
+		WeaklyTypedInput: true,
+		DecodeHook: mapstructure.ComposeDecodeHookFunc(
+			mapstructure.StringToTimeDurationHookFunc(),
+			mapstructure.StringToTimeHookFunc(time.RFC3339),
+			StringToDurationHookFunc(),
+			StringToByteSliceHookFunc(),
+		),
+		Result: &rawMsg,
+	}
+	decoder, decoderErr := mapstructure.NewDecoder(decoderConfig)
+	if decoderErr != nil {
+		panic(fmt.Errorf("error creating RawMsgSendGrant decoder: %v", decoderErr))
+	}
+	if err := decoder.Decode(msg); err != nil {
+		panic(fmt.Errorf("error decoding RawMsgSendGrant: %v", err))
+	}
+
+	if !msgCommonParams.TxSuccess {
+		params := ibc_model.MsgGrantParams{
+			MaybeSendGrant: &ibc_model.RawMsgSendGrant{
```

Comment:
I think here we could just use `&rawMsg`, no need to create the struct again lol

Many similar places below.. 🙏